lib/wiki_edits.rb

```diff
@@ -157,32 +117,23 @@ def self.notify_users(current_user, recipient_users, message)
     end
   end
 
-  def self.get_wiki_top_section(course_page_slug, current_user, talk_page=true)
-    tokens = get_tokens(current_user)
-    # if talk_page
-    #   course_prefix = Figaro.env.course_talk_prefix
-    # else
-    #   course_prefix = Figaro.env.course_prefix
-    # end
-    # page_title = "#{course_page_slug}"
-    puts course_page_slug
-    params = { action: 'parse',
-               page: course_page_slug,
-               section: '0',
-               prop: 'wikitext',
-               format: 'json' }
-
-    response = api_post params, tokens
-    puts response.body
-    response.body
-  end
+  def self.assignments_by_article(course, assignments, delete)
```

Comment:
My reasoning was that they are set from .update_assignments (where they do default to nil and false) — and this method is tightly coupled to .update_assignments — so it seemed clearer to rely on those being passed along. I trust your judgment, but I don't understand why setting defaults in both methods would be better. Can you explain your reasoning?